dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/persistence/file/batch/BatchFileOrchestrator.kt

```diff
@@ -313,40 +305,39 @@ internal class BatchFileOrchestrator(
         }
     }
 
-    private fun listSortedBatchFiles(): List<File> {
-        return rootDir.listFilesSafe(fileFilter, internalLogger).orEmpty().sorted()
+    private fun listBatchFiles(): List<File> {
+        return rootDir.listFilesSafe(fileFilter, internalLogger).orEmpty().toList()
     }
 
-    private val File.metadata: File
-        get() = File("${this.path}_metadata")
+    private fun listSortedBatchFiles(): List<File> {
+        // note: since it is using File#compareTo, lexicographical sorting will be used, meaning "10" comes before "9".
+        // but for our needs it is fine, because the moment when Unix timestamp adds one more digit will be in 2286.
+        return listBatchFiles().sorted()
+    }
 
     private fun canDoCleanup(): Boolean {
         return System.currentTimeMillis() - lastCleanupTimestamp > config.cleanupFrequencyThreshold
     }
 
+    private val File.metadata: File
+        get() = File("${this.path}_metadata")
+
+    private val File.isBatchFile: Boolean
+        get() = name.toLongOrNull() != null
+
+    private val List<File>.latestBatchFile: File?
+        get() = maxOrNull()
+
     // endregion
 
     // region FileFilter
 
-    internal inner class BatchFileFilter(private val internalLogger: InternalLogger) : FileFilter {
+    internal inner class BatchFileFilter : FileFilter {
         @Suppress("ReturnCount")
         override fun accept(file: File?): Boolean {
             if (file == null) return false
 
-            @Suppress("UnsafeThirdPartyFunctionCall") // value is not null
-            if (knownBatchFiles.get(file) != null) {
```

Comment:
isn't this dangerous ? I know the uploaded checks this set when trying to pick a file for uploading ?

Reply:
It is a cache, not a list of files locked for upload. Cache is not needed now as the performance without the cache is the same as with it (check the difference between tests 2, 3 and 4 once first optimization was applied).